addons/web/static/tests/views/list_tests.js

```diff
@@ -8536,6 +8554,102 @@ QUnit.module('Views', {
+        await testUtils.dom.triggerMouseEvent($('.daterangepicker:first .drp-calendar.right .available:contains("12")'), 'mousedown');
+
+
+        const $applyBtn = $('.daterangepicker:first .applyBtn');
+        assert.ok($applyBtn.length === 1 && !$applyBtn.disabled, 'Should only have 1 apply button in the daterangepicker and this button should be enabled.');
+
+        // daterange selected
+        const daterange = $('.daterangepicker:first .drp-selected').text();
+        let [dateStart, dateEnd] = daterange.split(' - ');
+        const dateFormat = "MM/DD/YYYY";
+
+        testUtils.mock.intercept(list, "field_changed", function (ev) {
+            const changes = ev.data.changes;
+            assert.ok(changes.hasOwnProperty('date_start') && changes.hasOwnProperty('date_end'));
+            const actualDateStart = changes.date_start.format(dateFormat);
+            const actualDateEnd = changes.date_end.format(dateFormat);
+            assert.strictEqual(actualDateStart, dateStart.trim());
+            assert.strictEqual(actualDateEnd, dateEnd.trim());
+        }, true);
+
+        // Apply the changes
+        await testUtils.dom.click($applyBtn);
+
+        assert.containsOnce(document.body, '.modal', 'The confirm dialog should appear to confirm the multi edition.');
+
+        // Valid the confirm dialog
+        await testUtils.dom.click($('.modal .btn-primary'));
+
+        assert.containsNone(document.body, '.modal', 'No confirm dialog should be visible in the dom.');
+
+        list.destroy();
+    });
+
+    QUnit.test('editable list: multi-edit field with daterange widget, user edits without using daterange picker', async function (assert) {
+        assert.expect(1);
+
+        const list = await createView({
+            View: ListView,
+            model: 'daterange',
+            data: this.data,
+            arch: `
+                <tree multi_edit="1">
+                    <field name="date_start" widget="daterange" options="{'related_end_date': 'date_end'}" />
+                    <field name="date_end" widget="daterange" options="{'related_start_date': 'date_start'}"/>
+                </tree>`,
+            session: {
+                getTZOffset: function () {
+                    return 330;
+                },
+            },
+        });
+        // Test manually edit the date without using the daterange picker
+        await testUtils.dom.click(list.$('thead .o_list_record_selector:first input'));
+        await testUtils.dom.click(list.$('.o_data_row:first .o_data_cell:eq(0) .o_field_date_range'));
+
+        // Change the date in the first datetime
+        await testUtils.fields.editInput(list.$('.o_data_row:first .o_data_cell:eq(0) .o_field_date_range'), "2021-04-01 11:00:00");
+        await testUtils.dom.click($('.modal .btn-primary'));
+
+        const newDate = moment.utc("2021-04-01", "YYYY-MM-DD");
+        assert.ok(Object.keys(list.model.localData).every(key => {
```

Comment:
hmmm, I know sometimes in tests we control inner variables like this, personnaly I'd prefer to control the dom (representative of the interactions wetween model, controller and renderer) rather than exposing implementation details of the model

Reply:
Do you prefer to test like the code below ?
```js
        const dateRangeFields = list.el.querySelectorAll('.o_data_cell .o_field_date_range');
        dateRangeFields.forEach(element => {
            if (element.getAttribute('name') === 'date_start') {
                assert.strictEqual(newDate.format('MM/DD/YYYY'), element.textContent);
            }
        });
```
instead of 
```js
assert.ok(Object.keys(list.model.localData).every(key => {
    ...
}));
```